pkg/statemachine/stateMachine.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package statemachine
+
+type StateDirection int64
+
+const (
+	Forward StateDirection = iota
+	Backward
+	Stop
+)
+
+type StateMachine struct {
+	index    int
+	states   []string
+	finished bool
+}
+
+func NewStateMachine(states []string) *StateMachine {
+	return &StateMachine{
+		states: states,
+	}
+}
+
+func (sm *StateMachine) CurrentState() string {
+	if sm.index < 0 || sm.index >= len(sm.states) {
+		return ""
```

Comment:
It already seems very improbable that this can happen at all, but I feel like in this case we might want to return an error instead? Or an error string constant at least?

Reply:
I preferred for CurrentState() to not return error, so the code has less boilerplate. The user should check for Running function to get this information (empty string means stopped).

Reply:
The `Running` function is in a way the same "boilerplate" as having a specific end state to query...therefore, being it basically the same I see no point in insisting for a change.